boa_engine/src/vm/code_block.rs

```diff
@@ -252,6 +250,33 @@ impl CodeBlock {
             .rev()
             .find(|(_, handler)| handler.contains(pc))
     }
+
+    pub(crate) fn constant_string_expect(&self, index: usize) -> JsString {
+        if let Constant::String(value) = &self.constants[index] {
+            return value.clone();
+        }
+
+        panic!("there should be a string constant at index {index}")
```

Comment:
Can there be `Panic` docs added to these functions so they pass future lints.

Also, a little nitpicky aside, I don't dislike naming the functions as `_expect`. I do think it's clear, but would it be worthwhile being more explicit with `_or_panic`? I'm fine with either/or but wanted to at least mention it.

Reply:
Maybe we should remove the postfix? I only added it because we have `.resolve_expect()` and `.environment_expect()` to be consistent with the naming. But there isn't a rust guideline neither does the `std` lib do this.

Reply:
+1 to the suggestion about removing the postfix. It's not common for Rust APIs to name panicking functions with a suffix. It's much more common to panic by default, then offer `try_*` variants for non-panicking functions.

Reply:
Yeah, I was sort of thinking about the `try_*` naming convention when I was looking at it. It's much more common for us to return a `JsResult` throughout the engine as a default. So instead of adding `try_*` everywhere, adding some sort of naming convention on the panicking functions in our case might be a bit better clarity-wise.

That being said, sticking to the typical naming convention and removing the suffix is also a valid option.